quickjs/src/main/jni/Context.cpp

```diff
@@ -658,7 +658,7 @@ Context::getJsToJavaConverter(JNIEnv* env, jclass type, bool boxed) {
   } else if (typeName == "int") {
     return [](Context* c, JNIEnv* env, const JSValueConst& v) {
       jvalue result;
-      if (!JS_IsInteger(v)) {
+      if (JS_VALUE_GET_TAG(v) != JS_TAG_INT) {
```

Comment:
Similar justification here except it's based on your own expectation of the type